In [49]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

In [50]:
df_origin = pd.read_csv("/content/drive/MyDrive/class/情報デザイン論/加工データ csv/user-1/user-1-nonull.csv")
df_origin.head()

Unnamed: 0  Recording timestamp  Computer timestamp       Sensor  \
0           0                    0          6405043858          NaN   
1           1                 8506          6405052364        Mouse   
2           2                18053          6405061911  Eye Tracker   
3           3                24203          6405068061        Mouse   
4           4                39987          6405083845        Mouse   

  Project name Export date Participant name Recording name Recording date  \
0          JD4  2025/07/18           user-1     Recording2     2025/07/18   
1          JD4  2025/07/18           user-1     Recording2     2025/07/18   
2          JD4  2025/07/18           user-1     Recording2     2025/07/18   
3          JD4  2025/07/18           user-1     Recording2     2025/07/18   
4          JD4  2025/07/18           user-1     Recording2     2025/07/18   

  Recording date UTC  ... Original Media height Eye movement type  \
0         2025/07/18  ...                   NaN               NaN   
1         2025/07/18  ...                   NaN               NaN   
2         2025/07/18  ...                   NaN      EyesNotFound   
3         2025/07/18  ...                   NaN      EyesNotFound   
4         2025/07/18  ...                   NaN      EyesNotFound   

   Eye movement event duration Eye movement type index Fixation point X  \
0                          NaN                     NaN              NaN   
1                          NaN                     NaN              NaN   
2                         47.0                     1.0              NaN   
3                         47.0                     1.0              NaN   
4                         47.0                     1.0              NaN   

  Fixation point Y  Fixation point X (MCSnorm)  Fixation point Y (MCSnorm)  \
0              NaN                         NaN                         NaN   
1              NaN                         NaN                         NaN   
2              NaN                         NaN                         NaN   
3              NaN                         NaN                         NaN   
4              NaN                         NaN                         NaN   

   Mouse position X  Mouse position Y  
0               NaN               NaN  
1            1447.0             650.0  
2               NaN               NaN  
3            1447.0             640.0  
4            1447.0             633.0  

[5 rows x 84 columns]

## カラムの整理

In [51]:
nunique_series = df_origin.nunique(dropna=False).to_string()
print(nunique_series)

Unnamed: 0                                    11888
Recording timestamp                           11875
Computer timestamp                            11875
Sensor                                            3
Project name                                      1
Export date                                       1
Participant name                                  1
Recording name                                    1
Recording date                                    1
Recording date UTC                                1
Recording start time                              1
Recording start time UTC                          1
Recording duration                                1
Timeline name                                     1
Recording Fixation filter name                    1
Recording software version                        1
Recording resolution height                       1
Recording resolution width                        1
Recording monitor latency                         1
Average cali

In [52]:
# 欠損値も含めてユニークな値の個数をカウント
nunique_series = df_origin.nunique(dropna=False)

# ユニークな値が1つのカラム名リスト
single_unique_cols = nunique_series[nunique_series == 1].index.tolist()

In [53]:
# カラム名とそのユニーク値を出力
for col in single_unique_cols:
    unique_val = df_origin[col].unique()
    print(f"{col}: {unique_val}")


Project name: ['JD4']
Export date: ['2025/07/18']
Participant name: ['user-1']
Recording name: ['Recording2']
Recording date: ['2025/07/18']
Recording date UTC: ['2025/07/18']
Recording start time: ['16:09:36.966']
Recording start time UTC: ['07:09:36.966']
Recording duration: [214788]
Timeline name: ['Timeline1']
Recording Fixation filter name: ['Tobii I-VT (Fixation)']
Recording software version: ['24.21.435']
Recording resolution height: [1080]
Recording resolution width: [1920]
Recording monitor latency: [10]
Average calibration accuracy (mm): [17.9]
Average calibration accuracy (degrees): [1.25]
Average calibration accuracy (pixels): [62]
Average validation accuracy (mm): [8]
Average validation precision SD (mm): [3.7]
Average validation precision RMS (mm): [4.9]
Average validation accuracy (degrees): [0.58]
Average validation precision SD (degrees): [0.27]
Average validation precision RMS (degrees): [0.35]
Average validation accuracy (pixels): [28]
Average validation precision SD

### カラムの要素が1つしかない変数の説明
|変数名(英語)|変数名(日本語)|中身|
|------------|--------------|----|
|Project name|プロジェクト名|`JD4`|
|Export date|エクスポート日| `2025/07/18`|
|Participant name| 参加者名|`user-1`|
|Recording name| 録画名|`Recording2`|
|Recording date|録画日(JSC) |`2025/07/18`|
|Recording date UTC| 録画日(UTC)|`2025/07/18`|
|Recording start time|録画開始時刻(JSC) |`16:09:36.966`|
|Recording start time UTC| 録画開始時刻(UTC) |`07:09:36.966`|
|Recording duration| 録画時間|`214788`ミリ秒|
|Timeline name| タイムライン名|`Timeline1`|
|Recording Fixation filter name| 録画の注視フィルタ名|`Tobii I-VT (Fixation)`|
|Recording software version| 録画ソフトのバージョン|`24.21.435`|
|Recording resolution height|録画解像度(高さ) |`1080`px|
|Recording resolution width|録画解像度(幅) | `1920`px|
|Recording monitor latency| モニター遅延|`10`ミリ秒|
|Average calibration accuracy | 平均キャリブレーション精度|`17.9`mm|
|Average calibration accuracy | 平均キャリブレーション精度|`1.25`degrees|
|Average calibration accuracy | 平均キャリブレーション精度|`62`pixels|
|Average validation accuracy | 平均バリデーション精度|`8`mm|
|Average validation precision SD | 平均バリデーション予測SD|`3.7`mm|
|Average validation precision RMS |平均バリデーション予測 RMS |`4.9`mm|
|Average validation accuracy |平均バリデーション精度| `0.58`degrees|
|Average validation precision SD |平均バリデーション予測 SD| `0.27`degrees|
|Average validation precision RMS |平均バリデーション予測 RMS| `0.35`degrees|
|Average validation accuracy |平均バリデーション精度| `28`pixels|
|Average validation precision SD |平均バリデーション予測SD| `13`pixels|
|Average validation precision RMS |平均バリデーション予測 RMS| `17`pixels|

In [54]:
# 削除した新しいデータフレーム
df_cleaned = df_origin.drop(columns=single_unique_cols)
df_cleaned.head()

Unnamed: 0  Recording timestamp  Computer timestamp       Sensor  \
0           0                    0          6405043858          NaN   
1           1                 8506          6405052364        Mouse   
2           2                18053          6405061911  Eye Tracker   
3           3                24203          6405068061        Mouse   
4           4                39987          6405083845        Mouse   

   Eyetracker timestamp           Event Event value  Gaze point X  \
0                   NaN  RecordingStart         NaN           NaN   
1                   NaN             NaN         NaN           NaN   
2          6.423402e+09             NaN         NaN           NaN   
3                   NaN             NaN         NaN           NaN   
4                   NaN             NaN         NaN           NaN   

   Gaze point Y  Gaze point left X  ...  Original Media height  \
0           NaN                NaN  ...                    NaN   
1           NaN                NaN  ...                    NaN   
2           NaN                NaN  ...                    NaN   
3           NaN                NaN  ...                    NaN   
4           NaN                NaN  ...                    NaN   

   Eye movement type  Eye movement event duration  Eye movement type index  \
0                NaN                          NaN                      NaN   
1                NaN                          NaN                      NaN   
2       EyesNotFound                         47.0                      1.0   
3       EyesNotFound                         47.0                      1.0   
4       EyesNotFound                         47.0                      1.0   

   Fixation point X  Fixation point Y  Fixation point X (MCSnorm)  \
0               NaN               NaN                         NaN   
1               NaN               NaN                         NaN   
2               NaN               NaN                         NaN   
3               NaN               NaN                         NaN   
4               NaN               NaN                         NaN   

   Fixation point Y (MCSnorm)  Mouse position X  Mouse position Y  
0                         NaN               NaN               NaN  
1                         NaN            1447.0             650.0  
2                         NaN               NaN               NaN  
3                         NaN            1447.0             640.0  
4                         NaN            1447.0             633.0  

[5 rows x 57 columns]

## ユニークな値を取り除いたデータフレーム

In [55]:
print(df_cleaned.nunique(dropna=False).to_string())

Unnamed: 0                             11888
Recording timestamp                    11875
Computer timestamp                     11875
Sensor                                     3
Eyetracker timestamp                   11595
Event                                      9
Event value                                9
Gaze point X                            1795
Gaze point Y                            1155
Gaze point left X                       1804
Gaze point left Y                       1171
Gaze point right X                      1810
Gaze point right Y                      1170
Gaze direction left X                   9515
Gaze direction left Y                   8652
Gaze direction left Z                   4534
Gaze direction right X                  9453
Gaze direction right Y                  8724
Gaze direction right Z                  4936
Pupil diameter left                     1141
Pupil diameter right                    1264
Pupil diameter filtered                 1166
Validity l

## Sensor毎にデータフレームを分割

In [56]:
# NaNを含む行
df_sensor_nan = df_cleaned[df_cleaned['Sensor'].isna()]

# "Mouse" の行
df_sensor_mouse = df_cleaned[df_cleaned['Sensor'] == "Mouse"]

# "Eye Tracker" の行
df_sensor_eye = df_cleaned[df_cleaned['Sensor'] == "Eye Tracker"]


### 問題の切り替わりの時間を取得する

In [57]:
df_sensor_nan

Unnamed: 0  Recording timestamp  Computer timestamp Sensor  \
0               0                    0          6405043858    NaN   
209           209              2980387          6408024245    NaN   
467           467              6104332          6411148190    NaN   
479           479              6250365          6411294223    NaN   
1995         1995             49337171          6454381029    NaN   
1999         1999             49427175          6454471033    NaN   
2029         2029             49851643          6454895501    NaN   
2030         2030             49851643          6454895501    NaN   
2337         2337             54849574          6459893432    NaN   
2338         2338             54849574          6459893432    NaN   
4229         4229             86315930          6491359788    NaN   
4232         4232             86348807          6491392665    NaN   
4233         4233             86348807          6491392665    NaN   
4534         4534             91349501          6496393359    NaN   
4535         4535             91349501          6496393359    NaN   
6966         6966            131776861          6536820719    NaN   
6969         6969            131812950          6536856808    NaN   
6970         6970            131812950          6536856808    NaN   
7260         7260            136804614          6541848472    NaN   
7261         7261            136804614          6541848472    NaN   
9365         9365            171826976          6576870834    NaN   
9368         9368            171870866          6576914724    NaN   
9369         9369            171870866          6576914724    NaN   
9622         9622            176865961          6581909819    NaN   
9623         9623            176865961          6581909819    NaN   
11587       11587            209538006          6614581864    NaN   
11590       11590            209566796          6614610654    NaN   
11591       11591            209566796          6614610654    NaN   
11883       11883            214573898          6619617756    NaN   
11887       11887            214788401          6619832259    NaN   

       Eyetracker timestamp                          Event Event value  \
0                       NaN                 RecordingStart         NaN   
209                     NaN  Eye tracker Calibration start         NaN   
467                     NaN                     MouseEvent  Down, Left   
479                     NaN                     MouseEvent    Up, Left   
1995                    NaN                     MouseEvent  Down, Left   
1999                    NaN                     MouseEvent    Up, Left   
2029                    NaN    Eye tracker Calibration end         NaN   
2030                    NaN             ImageStimulusStart      blank2   
2337                    NaN               ImageStimulusEnd      blank2   
2338                    NaN             ImageStimulusStart  視線誘導問題.002   
4229                    NaN                  KeyboardEvent       Space   
4232                    NaN               ImageStimulusEnd  視線誘導問題.002   
4233                    NaN             ImageStimulusStart      blank2   
4534                    NaN               ImageStimulusEnd      blank2   
4535                    NaN             ImageStimulusStart  視線誘導問題.001   
6966                    NaN                  KeyboardEvent       Space   
6969                    NaN               ImageStimulusEnd  視線誘導問題.001   
6970                    NaN             ImageStimulusStart      blank2   
7260                    NaN               ImageStimulusEnd      blank2   
7261                    NaN             ImageStimulusStart  視線誘導問題.003   
9365                    NaN                  KeyboardEvent       Space   
9368                    NaN               ImageStimulusEnd  視線誘導問題.003   
9369                    NaN             ImageStimulusStart      blank2   
9622                    NaN               ImageStimulusEnd      blank2   
9623             

In [58]:
# "視線誘導問題." を含む行を抽出
df_filtered = df_sensor_nan[df_sensor_nan["Event value"].astype(str).str.contains("視線誘導問題.")].copy()

# 結果の確認（先頭5行を表示）
df_filtered.loc[:,['Recording timestamp', 'Computer timestamp', 'Event value']]
print(df_filtered.loc[:,['Recording timestamp', 'Computer timestamp', 'Event value']])

       Recording timestamp  Computer timestamp Event value
2338              54849574          6459893432  視線誘導問題.002
4232              86348807          6491392665  視線誘導問題.002
4535              91349501          6496393359  視線誘導問題.001
6969             131812950          6536856808  視線誘導問題.001
7261             136804614          6541848472  視線誘導問題.003
9368             171870866          6576914724  視線誘導問題.003
9623             176865961          6581909819  視線誘導問題.004
11590            209566796          6614610654  視線誘導問題.004


# 各問題の開始時刻
|       |Recording timestamp|  Computer timestamp| Event value|
|-------|-------------------|--------------------|------------|
|2338              |54849574          |6459893432  |視線誘導問題.002|
|4232              |86348807          |6491392665  |視線誘導問題.002|
|4535              |91349501          |6496393359  |視線誘導問題.001|
|6969             |131812950          |6536856808  |視線誘導問題.001|
|7261             |136804614          |6541848472  |視線誘導問題.003|
|9368             |171870866          |6576914724  |視線誘導問題.003|
|9623             |176865961          |6581909819  |視線誘導問題.004|
|11590            |209566796          |6614610654  |視線誘導問題.004|


In [59]:
df_filtered.loc[:,['Recording timestamp', 'Computer timestamp', 'Event value']].to_csv("/content/drive/MyDrive/class/情報デザイン論/加工データ csv/user-1/start-stop-timing.csv")

In [61]:
df_sensor_mouse

Unnamed: 0  Recording timestamp  Computer timestamp Sensor  \
1              1                 8506          6405052364  Mouse   
3              3                24203          6405068061  Mouse   
4              4                39987          6405083845  Mouse   
6              6                55832          6405099690  Mouse   
8              8                72061          6405115919  Mouse   
...          ...                  ...                 ...    ...   
2044        2044             50001115          6455044973  Mouse   
2046        2046             50017091          6455060949  Mouse   
2048        2048             50033102          6455076960  Mouse   
2050        2050             50051082          6455094940  Mouse   
2052        2052             50067082          6455110940  Mouse   

      Eyetracker timestamp Event Event value  Gaze point X  Gaze point Y  \
1                      NaN   NaN         NaN           NaN           NaN   
3                      NaN   NaN         NaN           NaN           NaN   
4                      NaN   NaN         NaN           NaN           NaN   
6                      NaN   NaN         NaN           NaN           NaN   
8                      NaN   NaN         NaN           NaN           NaN   
...                    ...   ...         ...           ...           ...   
2044                   NaN   NaN         NaN           NaN           NaN   
2046                   NaN   NaN         NaN           NaN           NaN   
2048                   NaN   NaN         NaN           NaN           NaN   
2050                   NaN   NaN         NaN           NaN           NaN   
2052                   NaN   NaN         NaN           NaN           NaN   

      Gaze point left X  ...  Original Media height  Eye movement type  \
1                   NaN  ...                    NaN                NaN   
3                   NaN  ...                    NaN       EyesNotFound   
4                   NaN  ...                    NaN       EyesNotFound   
6                   NaN  ...                    NaN       EyesNotFound   
8                   NaN  ...                    NaN       Unclassified   
...                 ...  ...                    ...                ...   
2044                NaN  ...                 1080.0           Fixation   
2046                NaN  ...                 1080.0           Fixation   
2048                NaN  ...                 1080.0           Fixation   
2050                NaN  ...                 1080.0           Fixation   
2052                NaN  ...                 1080.0           Fixation   

      Eye movement event duration  Eye movement type index  Fixation point X  \
1                             NaN                      NaN               NaN   
3                            47.0                      1.0               NaN   
4                            47.0                      1.0               NaN   
6                            47.0                      1.0               NaN   
8                            17.0                      1.0               NaN   
...                           ...                      ...               ...   
2044                        133.0                     66.0             950.0   
2046                        133.0                     66.0             950.0   
2048                        133.0                     66.0             950.0   
2050                        133.0                     66.0             950.0   
2052                        133.0                     66.0             950.0   

      Fixation point Y  Fixation point X (MCSnorm)  \
1                  NaN                         NaN   
3                  NaN                         NaN   
4                  NaN                         NaN   
6                  NaN                         NaN   
8                  NaN                         NaN   
...                ...                         ...   
2044             438.0                      0.49

In [60]:
df_sensor_eye

Unnamed: 0  Recording timestamp  Computer timestamp       Sensor  \
2               2                18053          6405061911  Eye Tracker   
5               5                48415          6405092273  Eye Tracker   
7               7                65758          6405109616  Eye Tracker   
9               9                82399          6405126257  Eye Tracker   
11             11                98589          6405142447  Eye Tracker   
...           ...                  ...                 ...          ...   
11881       11881            214547272          6619591130  Eye Tracker   
11882       11882            214563876          6619607734  Eye Tracker   
11884       11884            214580535          6619624393  Eye Tracker   
11885       11885            214597195          6619641053  Eye Tracker   
11886       11886            214613845          6619657703  Eye Tracker   

       Eyetracker timestamp Event Event value  Gaze point X  Gaze point Y  \
2              6.423402e+09   NaN         NaN           NaN           NaN   
5              6.423432e+09   NaN         NaN           NaN           NaN   
7              6.423450e+09   NaN         NaN         495.0         626.0   
9              6.423466e+09   NaN         NaN           NaN           NaN   
11             6.423482e+09   NaN         NaN           NaN           NaN   
...                     ...   ...         ...           ...           ...   
11881          6.637918e+09   NaN         NaN        1176.0        1029.0   
11882          6.637934e+09   NaN         NaN        1173.0        1026.0   
11884          6.637951e+09   NaN         NaN        1179.0        1009.0   
11885          6.637968e+09   NaN         NaN        1189.0        1003.0   
11886          6.637984e+09   NaN         NaN        1182.0        1002.0   

       Gaze point left X  ...  Original Media height  Eye movement type  \
2                    NaN  ...                    NaN       EyesNotFound   
5                    NaN  ...                    NaN       EyesNotFound   
7                    NaN  ...                    NaN       Unclassified   
9                    NaN  ...                    NaN       EyesNotFound   
11                   NaN  ...                    NaN       EyesNotFound   
...                  ...  ...                    ...                ...   
11881             1166.0  ...                 1080.0            Saccade   
11882             1179.0  ...                 1080.0           Fixation   
11884             1167.0  ...                    NaN           Fixation   
11885             1190.0  ...                    NaN           Fixation   
11886             1200.0  ...                    NaN           Fixation   

       Eye movement event duration  Eye movement type index  Fixation point X  \
2                             47.0                      1.0               NaN   
5                             47.0                      1.0               NaN   
7                             17.0                      1.0               NaN   
9                            117.0                      2.0               NaN   
11                           117.0                      2.0               NaN   
...                            ...                      ...               ...   
11881                         50.0                    885.0               NaN   
11882                         67.0                    644.0            1180.0   
11884                         67.0                    644.0            1180.0   
11885                         67.0                    644.0            1180.0   
11886                         67.0                    644.0            1180.0   

       Fixation point Y  Fixation point X (MCSnorm)  \
2                   NaN                         NaN   
5                   NaN                         NaN   
7                   NaN                         NaN   
9                   NaN                         NaN   
11                  NaN         

In [62]:
df_cleaned_ver2 = df_cleaned[df_cleaned['Sensor'] != "Mouse"]
nunique_series_ver2 = df_cleaned_ver2.nunique(dropna=False)
single_unique_cols_ver2 = nunique_series_ver2[nunique_series_ver2 == 1].index.tolist()
# カラム名とそのユニーク値を出力
for col in single_unique_cols_ver2:
    unique_val = df_cleaned_ver2[col].unique()
    print(f"{col}: {unique_val}")

Mouse position X: [nan]
Mouse position Y: [nan]


In [63]:
df_cleaned_ver2 = df_cleaned_ver2.drop(columns=single_unique_cols_ver2)

In [64]:
print(df_cleaned_ver2.nunique(dropna=False).to_string())

Unnamed: 0                             11624
Recording timestamp                    11615
Computer timestamp                     11615
Sensor                                     2
Eyetracker timestamp                   11595
Event                                      9
Event value                                9
Gaze point X                            1795
Gaze point Y                            1155
Gaze point left X                       1804
Gaze point left Y                       1171
Gaze point right X                      1810
Gaze point right Y                      1170
Gaze direction left X                   9515
Gaze direction left Y                   8652
Gaze direction left Z                   4534
Gaze direction right X                  9453
Gaze direction right Y                  8724
Gaze direction right Z                  4936
Pupil diameter left                     1141
Pupil diameter right                    1264
Pupil diameter filtered                 1166
Validity l

In [65]:
df_cleaned_ver2.head()

Unnamed: 0  Recording timestamp  Computer timestamp       Sensor  \
0           0                    0          6405043858          NaN   
2           2                18053          6405061911  Eye Tracker   
5           5                48415          6405092273  Eye Tracker   
7           7                65758          6405109616  Eye Tracker   
9           9                82399          6405126257  Eye Tracker   

   Eyetracker timestamp           Event Event value  Gaze point X  \
0                   NaN  RecordingStart         NaN           NaN   
2          6.423402e+09             NaN         NaN           NaN   
5          6.423432e+09             NaN         NaN           NaN   
7          6.423450e+09             NaN         NaN         495.0   
9          6.423466e+09             NaN         NaN           NaN   

   Gaze point Y  Gaze point left X  ...  Presented Media position Y (DACSpx)  \
0           NaN                NaN  ...                                  NaN   
2           NaN                NaN  ...                                  NaN   
5           NaN                NaN  ...                                  NaN   
7         626.0                NaN  ...                                  NaN   
9           NaN                NaN  ...                                  NaN   

   Original Media width  Original Media height  Eye movement type  \
0                   NaN                    NaN                NaN   
2                   NaN                    NaN       EyesNotFound   
5                   NaN                    NaN       EyesNotFound   
7                   NaN                    NaN       Unclassified   
9                   NaN                    NaN       EyesNotFound   

   Eye movement event duration  Eye movement type index  Fixation point X  \
0                          NaN                      NaN               NaN   
2                         47.0                      1.0               NaN   
5                         47.0                      1.0               NaN   
7                         17.0                      1.0               NaN   
9                        117.0                      2.0               NaN   

   Fixation point Y  Fixation point X (MCSnorm)  Fixation point Y (MCSnorm)  
0               NaN                         NaN                         NaN  
2               NaN                         NaN                         NaN  
5               NaN                         NaN                         NaN  
7               NaN                         NaN                         NaN  
9               NaN                         NaN                         NaN  

[5 rows x 55 columns]

In [66]:
# 欠損値も含めてユニークな値の個数をカウント
nunique_series_ver3 = df_cleaned_ver2.nunique(dropna=False)

# ユニークな値が1つのカラム名リスト
single_unique_cols_ver3 = nunique_series_ver3[nunique_series_ver3 == 2].index.tolist()

# カラム名とそのユニーク値を出力
for col in single_unique_cols_ver3:
    unique_val = df_cleaned_ver2[col].unique()
    print(f"{col}: {unique_val}")


Sensor: [nan 'Eye Tracker']
Presented Media width: [  nan 1920.]
Presented Media height: [  nan 1080.]
Presented Media position X (DACSpx): [nan  0.]
Presented Media position Y (DACSpx): [nan  0.]
Original Media width: [  nan 1920.]
Original Media height: [  nan 1080.]


### Sensor=Mouse以外に着目した後に要素が少ない特徴量について
|変数名(英語)|変数名(日本語)|中身|
|------------|--------------|----|
|Mouse position X| マウス位置 X|`nan`|
|Mouse position Y| マウス位置 Y|`nan`|
|Sensor| センサー|`nan`, `Eye Tracker`|
|Presented Media width|提示されたメディアの幅|`nan` `1920`|
|Presented Media height|提示されたメディアの高さ|`nan`, `1080`|
|Presented Media position X (DACSpx)| 提示されたメディア位置 X|`nan`,  `0`|
|Presented Media position Y (DACSpx)| 提示されたメディアの位置 Y|`nan`,  `0`|
|Original Media width| オリジナルメディアの幅|`nan`, `1920`|
|Original Media height| オリジナルメディアの高さ|`nan`,`1080`|

In [67]:
df_sensor_eye

Unnamed: 0  Recording timestamp  Computer timestamp       Sensor  \
2               2                18053          6405061911  Eye Tracker   
5               5                48415          6405092273  Eye Tracker   
7               7                65758          6405109616  Eye Tracker   
9               9                82399          6405126257  Eye Tracker   
11             11                98589          6405142447  Eye Tracker   
...           ...                  ...                 ...          ...   
11881       11881            214547272          6619591130  Eye Tracker   
11882       11882            214563876          6619607734  Eye Tracker   
11884       11884            214580535          6619624393  Eye Tracker   
11885       11885            214597195          6619641053  Eye Tracker   
11886       11886            214613845          6619657703  Eye Tracker   

       Eyetracker timestamp Event Event value  Gaze point X  Gaze point Y  \
2              6.423402e+09   NaN         NaN           NaN           NaN   
5              6.423432e+09   NaN         NaN           NaN           NaN   
7              6.423450e+09   NaN         NaN         495.0         626.0   
9              6.423466e+09   NaN         NaN           NaN           NaN   
11             6.423482e+09   NaN         NaN           NaN           NaN   
...                     ...   ...         ...           ...           ...   
11881          6.637918e+09   NaN         NaN        1176.0        1029.0   
11882          6.637934e+09   NaN         NaN        1173.0        1026.0   
11884          6.637951e+09   NaN         NaN        1179.0        1009.0   
11885          6.637968e+09   NaN         NaN        1189.0        1003.0   
11886          6.637984e+09   NaN         NaN        1182.0        1002.0   

       Gaze point left X  ...  Original Media height  Eye movement type  \
2                    NaN  ...                    NaN       EyesNotFound   
5                    NaN  ...                    NaN       EyesNotFound   
7                    NaN  ...                    NaN       Unclassified   
9                    NaN  ...                    NaN       EyesNotFound   
11                   NaN  ...                    NaN       EyesNotFound   
...                  ...  ...                    ...                ...   
11881             1166.0  ...                 1080.0            Saccade   
11882             1179.0  ...                 1080.0           Fixation   
11884             1167.0  ...                    NaN           Fixation   
11885             1190.0  ...                    NaN           Fixation   
11886             1200.0  ...                    NaN           Fixation   

       Eye movement event duration  Eye movement type index  Fixation point X  \
2                             47.0                      1.0               NaN   
5                             47.0                      1.0               NaN   
7                             17.0                      1.0               NaN   
9                            117.0                      2.0               NaN   
11                           117.0                      2.0               NaN   
...                            ...                      ...               ...   
11881                         50.0                    885.0               NaN   
11882                         67.0                    644.0            1180.0   
11884                         67.0                    644.0            1180.0   
11885                         67.0                    644.0            1180.0   
11886                         67.0                    644.0            1180.0   

       Fixation point Y  Fixation point X (MCSnorm)  \
2                   NaN                         NaN   
5                   NaN                         NaN   
7                   NaN                         NaN   
9                   NaN                         NaN   
11                  NaN         

In [68]:
# 結果を辞書に格納
extracted_data = {}

targets = df_filtered["Event value"].unique()
for target in targets:
    timestamps = df_filtered[df_filtered["Event value"] == target]["Recording timestamp"].iloc[0:2]
    if len(timestamps) < 2:
        continue  # start/endが不足している場合スキップ

    start, end = timestamps.iloc[0], timestamps.iloc[1]

    extracted = df_sensor_eye[
        (df_sensor_eye["Recording timestamp"] >= start) &
        (df_sensor_eye["Recording timestamp"] <= end)
    ].copy()

    extracted_data[target] = extracted  # 日本語のキーOK

In [69]:
extracted_data["視線誘導問題.002"].head()

Unnamed: 0  Recording timestamp  Computer timestamp       Sensor  \
2339        2339             54863940          6459907798  Eye Tracker   
2340        2340             54880094          6459923952  Eye Tracker   
2341        2341             54897334          6459941192  Eye Tracker   
2342        2342             54913956          6459957814  Eye Tracker   
2343        2343             54930018          6459973876  Eye Tracker   

      Eyetracker timestamp Event Event value  Gaze point X  Gaze point Y  \
2339          6.478244e+09   NaN         NaN         930.0        1064.0   
2340          6.478260e+09   NaN         NaN         959.0        1065.0   
2341          6.478278e+09   NaN         NaN         947.0        1074.0   
2342          6.478294e+09   NaN         NaN         946.0        1086.0   
2343          6.478310e+09   NaN         NaN         939.0        1072.0   

      Gaze point left X  ...  Original Media height  Eye movement type  \
2339              901.0  ...                 1080.0           Fixation   
2340              946.0  ...                 1080.0           Fixation   
2341              945.0  ...                 1080.0           Fixation   
2342              928.0  ...                 1080.0           Fixation   
2343              928.0  ...                 1080.0           Fixation   

      Eye movement event duration  Eye movement type index  Fixation point X  \
2339                        150.0                     79.0             944.0   
2340                        150.0                     79.0             944.0   
2341                        150.0                     79.0             944.0   
2342                        150.0                     79.0             944.0   
2343                        150.0                     79.0             944.0   

      Fixation point Y  Fixation point X (MCSnorm)  \
2339            1073.0                      0.4915   
2340            1073.0                      0.4915   
2341            1073.0                      0.4915   
2342            1073.0                      0.4915   
2343            1073.0                      0.4915   

      Fixation point Y (MCSnorm)  Mouse position X  Mouse position Y  
2339                      0.9936               NaN               NaN  
2340                      0.9936               NaN               NaN  
2341                      0.9936               NaN               NaN  
2342                      0.9936               NaN               NaN  
2343                      0.9936               NaN               NaN  

[5 rows x 57 columns]

In [70]:
print(extracted_data["視線誘導問題.002"].nunique(dropna=False).to_string())

Unnamed: 0                             1892
Recording timestamp                    1892
Computer timestamp                     1892
Sensor                                    1
Eyetracker timestamp                   1892
Event                                     1
Event value                               1
Gaze point X                            980
Gaze point Y                            493
Gaze point left X                       967
Gaze point left Y                       507
Gaze point right X                      974
Gaze point right Y                      495
Gaze direction left X                  1686
Gaze direction left Y                  1560
Gaze direction left Z                  1397
Gaze direction right X                 1708
Gaze direction right Y                 1600
Gaze direction right Z                 1396
Pupil diameter left                     503
Pupil diameter right                    527
Pupil diameter filtered                 484
Validity left                   

In [71]:
extracted_data["視線誘導問題.001"].head()

Unnamed: 0  Recording timestamp  Computer timestamp       Sensor  \
4536        4536             91353604          6496397462  Eye Tracker   
4537        4537             91370512          6496414370  Eye Tracker   
4538        4538             91386638          6496430496  Eye Tracker   
4539        4539             91403573          6496447431  Eye Tracker   
4540        4540             91420430          6496464288  Eye Tracker   

      Eyetracker timestamp Event Event value  Gaze point X  Gaze point Y  \
4536          6.514732e+09   NaN         NaN        1616.0         575.0   
4537          6.514749e+09   NaN         NaN        1616.0         623.0   
4538          6.514765e+09   NaN         NaN        1573.0         647.0   
4539          6.514782e+09   NaN         NaN        1611.0         570.0   
4540          6.514798e+09   NaN         NaN        1607.0         592.0   

      Gaze point left X  ...  Original Media height  Eye movement type  \
4536             1646.0  ...                 1080.0           Fixation   
4537             1665.0  ...                 1080.0            Saccade   
4538             1613.0  ...                 1080.0       Unclassified   
4539             1655.0  ...                 1080.0            Saccade   
4540             1650.0  ...                 1080.0           Fixation   

      Eye movement event duration  Eye movement type index  Fixation point X  \
4536                         67.0                    204.0            1598.0   
4537                         17.0                    290.0               NaN   
4538                         17.0                     80.0               NaN   
4539                         17.0                    291.0               NaN   
4540                        183.0                    205.0            1592.0   

      Fixation point Y  Fixation point X (MCSnorm)  \
4536             600.0                      0.8321   
4537               NaN                         NaN   
4538               NaN                         NaN   
4539               NaN                         NaN   
4540             602.0                      0.8291   

      Fixation point Y (MCSnorm)  Mouse position X  Mouse position Y  
4536                      0.5560               NaN               NaN  
4537                         NaN               NaN               NaN  
4538                         NaN               NaN               NaN  
4539                         NaN               NaN               NaN  
4540                      0.5575               NaN               NaN  

[5 rows x 57 columns]

In [72]:
extracted_data["視線誘導問題.003"].head()

Unnamed: 0  Recording timestamp  Computer timestamp       Sensor  \
7262        7262            136805014          6541848872  Eye Tracker   
7263        7263            136821691          6541865549  Eye Tracker   
7264        7264            136838382          6541882240  Eye Tracker   
7265        7265            136854931          6541898789  Eye Tracker   
7266        7266            136871614          6541915472  Eye Tracker   

      Eyetracker timestamp Event Event value  Gaze point X  Gaze point Y  \
7262          6.560180e+09   NaN         NaN         262.0        1106.0   
7263          6.560197e+09   NaN         NaN         276.0        1098.0   
7264          6.560214e+09   NaN         NaN         278.0        1101.0   
7265          6.560230e+09   NaN         NaN         267.0        1113.0   
7266          6.560247e+09   NaN         NaN         268.0        1107.0   

      Gaze point left X  ...  Original Media height  Eye movement type  \
7262              247.0  ...                 1080.0           Fixation   
7263              260.0  ...                 1080.0           Fixation   
7264              265.0  ...                 1080.0           Fixation   
7265              264.0  ...                 1080.0           Fixation   
7266              279.0  ...                 1080.0           Fixation   

      Eye movement event duration  Eye movement type index  Fixation point X  \
7262                        233.0                    378.0             268.0   
7263                        233.0                    378.0             268.0   
7264                        233.0                    378.0             268.0   
7265                        233.0                    378.0             268.0   
7266                        233.0                    378.0             268.0   

      Fixation point Y  Fixation point X (MCSnorm)  \
7262            1114.0                         NaN   
7263            1114.0                         NaN   
7264            1114.0                         NaN   
7265            1114.0                         NaN   
7266            1114.0                         NaN   

      Fixation point Y (MCSnorm)  Mouse position X  Mouse position Y  
7262                         NaN               NaN               NaN  
7263                         NaN               NaN               NaN  
7264                         NaN               NaN               NaN  
7265                         NaN               NaN               NaN  
7266                         NaN               NaN               NaN  

[5 rows x 57 columns]

In [73]:
extracted_data["視線誘導問題.004"].head()

Unnamed: 0  Recording timestamp  Computer timestamp       Sensor  \
9624        9624            176875557          6581919415  Eye Tracker   
9625        9625            176891913          6581935771  Eye Tracker   
9626        9626            176908655          6581952513  Eye Tracker   
9627        9627            176925379          6581969237  Eye Tracker   
9628        9628            176941836          6581985694  Eye Tracker   

      Eyetracker timestamp Event Event value  Gaze point X  Gaze point Y  \
9624          6.600248e+09   NaN         NaN         423.0        1018.0   
9625          6.600265e+09   NaN         NaN         329.0        1081.0   
9626          6.600281e+09   NaN         NaN         346.0        1053.0   
9627          6.600298e+09   NaN         NaN         340.0        1045.0   
9628          6.600315e+09   NaN         NaN         242.0        1095.0   

      Gaze point left X  ...  Original Media height  Eye movement type  \
9624              473.0  ...                 1080.0       Unclassified   
9625                NaN  ...                 1080.0            Saccade   
9626              401.0  ...                 1080.0            Saccade   
9627              397.0  ...                 1080.0       Unclassified   
9628              246.0  ...                 1080.0            Saccade   

      Eye movement event duration  Eye movement type index  Fixation point X  \
9624                         17.0                    172.0               NaN   
9625                         33.0                    688.0               NaN   
9626                         33.0                    688.0               NaN   
9627                         17.0                    173.0               NaN   
9628                         17.0                    689.0               NaN   

      Fixation point Y  Fixation point X (MCSnorm)  \
9624               NaN                         NaN   
9625               NaN                         NaN   
9626               NaN                         NaN   
9627               NaN                         NaN   
9628               NaN                         NaN   

      Fixation point Y (MCSnorm)  Mouse position X  Mouse position Y  
9624                         NaN               NaN               NaN  
9625                         NaN               NaN               NaN  
9626                         NaN               NaN               NaN  
9627                         NaN               NaN               NaN  
9628                         NaN               NaN               NaN  

[5 rows x 57 columns]

In [74]:
for key in extracted_data.keys():
    print(f"Key: {key}")
    df = extracted_data[key]
    nunique_series = df.nunique(dropna=False)
    single_unique_cols = nunique_series[nunique_series == 1].index.tolist()
    for col in single_unique_cols:
        unique_val = df[col].unique()
        print(f"  {col}: {unique_val}")

Key: 視線誘導問題.002
  Sensor: ['Eye Tracker']
  Event: [nan]
  Event value: [nan]
  Presented Stimulus name: ['視線誘導問題.002']
  Presented Media name: ['視線誘導問題.002.jpeg']
  Presented Media width: [1920.]
  Presented Media height: [1080.]
  Presented Media position X (DACSpx): [0.]
  Presented Media position Y (DACSpx): [0.]
  Original Media width: [1920.]
  Original Media height: [1080.]
  Mouse position X: [nan]
  Mouse position Y: [nan]
Key: 視線誘導問題.001
  Sensor: ['Eye Tracker']
  Event: [nan]
  Event value: [nan]
  Presented Stimulus name: ['視線誘導問題.001']
  Presented Media name: ['視線誘導問題.001.jpeg']
  Presented Media width: [1920.]
  Presented Media height: [1080.]
  Presented Media position X (DACSpx): [0.]
  Presented Media position Y (DACSpx): [0.]
  Original Media width: [1920.]
  Original Media height: [1080.]
  Mouse position X: [nan]
  Mouse position Y: [nan]
Key: 視線誘導問題.003
  Sensor: ['Eye Tracker']
  Event: [nan]
  Event value: [nan]
  Presented Stimulus name: ['視線誘導問題.003']
  Presen

In [75]:
# 空の辞書に結果を格納
cleaned_data = {}

for key, df in extracted_data.items():
    nunique_series = df.nunique(dropna=False)
    single_unique_cols = nunique_series[nunique_series == 1].index.tolist()

    # そのカラムを除外して新しいデータフレームを作成
    df_cleaned = df.drop(columns=single_unique_cols).copy()

    cleaned_data[key] = df_cleaned

    print(f"{key}: Removed {len(single_unique_cols)} columns -> Remaining {df_cleaned.shape[1]} columns")


視線誘導問題.002: Removed 13 columns -> Remaining 44 columns
視線誘導問題.001: Removed 13 columns -> Remaining 44 columns
視線誘導問題.003: Removed 13 columns -> Remaining 44 columns
視線誘導問題.004: Removed 13 columns -> Remaining 44 columns


### カラムの要素が1つしかない変数の説明 写真2
|変数名(英語)|変数名(日本語)|中身|
|------------|--------------|----|
|Project name|プロジェクト名|`JD4`|
|Export date|エクスポート日| `2025/07/18`|
|Participant name| 参加者名|`user-1`|
|Recording name| 録画名|`Recording2`|
|Recording date|録画日(JSC) |`2025/07/18`|
|Recording date UTC| 録画日(UTC)|`2025/07/18`|
|Recording start time|録画開始時刻(JSC) |`16:09:36.966`|
|Recording start time UTC| 録画開始時刻(UTC) |`07:09:36.966`|
|Recording duration| 録画時間|`214788`ミリ秒|
|Timeline name| タイムライン名|`Timeline1`|
|Recording Fixation filter name| 録画の注視フィルタ名|`Tobii I-VT (Fixation)`|
|Recording software version| 録画ソフトのバージョン|`24.21.435`|
|Recording resolution height|録画解像度(高さ) |`1080`px|
|Recording resolution width|録画解像度(幅) | `1920`px|
|Recording monitor latency| モニター遅延|`10`ミリ秒|
|Average calibration accuracy | 平均キャリブレーション精度|`17.9`mm|
|Average calibration accuracy | 平均キャリブレーション精度|`1.25`degrees|
|Average calibration accuracy | 平均キャリブレーション精度|`62`pixels|
|Average validation accuracy | 平均バリデーション精度|`8`mm|
|Average validation precision SD | 平均バリデーション予測SD|`3.7`mm|
|Average validation precision RMS |平均バリデーション予測 RMS |`4.9`mm|
|Average validation accuracy |平均バリデーション精度| `0.58`degrees|
|Average validation precision SD |平均バリデーション予測 SD| `0.27`degrees|
|Average validation precision RMS |平均バリデーション予測 RMS| `0.35`degrees|
|Average validation accuracy |平均バリデーション精度| `28`pixels|
|Average validation precision SD |平均バリデーション予測SD| `13`pixels|
|Average validation precision RMS |平均バリデーション予測 RMS| `17`pixels|
|---------------------------------|--------------------------|-----------|
|  Sensor|センサー| `Eye Tracker`|
|  Event|イベント |`nan`|
|Event value|イベントの中身| `nan`|
|Presented Stimulus name|提示された刺激名| `視線誘導問題.002`|
|Presented Media name|提示されたメディア名| `視線誘導問題.002.jpeg`|
|Presented Media width|提示されたメディア幅|`1920`px|
|Presented Media height|提示されたメディア高さ|`1080`px|
|Presented Media position X (DACSpx)|提示されたメディア位置 X|`0`|
|Presented Media position Y (DACSpx)|提示されたメディア位置 Y|`0`|
|Original Media width|オリジナルメディアの幅|`1920`px|
|Original Media height |オリジナルメディアの高さ|`1080`px|
|Mouse position X|マウス位置 X|`nan`|
|Mouse position Y|マウス位置 Y|  `nan`|


In [77]:
cleaned_data["視線誘導問題.002"]

Unnamed: 0  Recording timestamp  Computer timestamp  \
2339        2339             54863940          6459907798   
2340        2340             54880094          6459923952   
2341        2341             54897334          6459941192   
2342        2342             54913956          6459957814   
2343        2343             54930018          6459973876   
...          ...                  ...                 ...   
4226        4226             86278030          6491321888   
4227        4227             86294487          6491338345   
4228        4228             86311367          6491355225   
4230        4230             86328045          6491371903   
4231        4231             86344410          6491388268   

      Eyetracker timestamp  Gaze point X  Gaze point Y  Gaze point left X  \
2339          6.478244e+09         930.0        1064.0              901.0   
2340          6.478260e+09         959.0        1065.0              946.0   
2341          6.478278e+09         947.0        1074.0              945.0   
2342          6.478294e+09         946.0        1086.0              928.0   
2343          6.478310e+09         939.0        1072.0              928.0   
...                    ...           ...           ...                ...   
4226          6.509656e+09         709.0         851.0              691.0   
4227          6.509673e+09         695.0         856.0              687.0   
4228          6.509690e+09         693.0         868.0              687.0   
4230          6.509706e+09         699.0         854.0              674.0   
4231          6.509723e+09         691.0         862.0              672.0   

      Gaze point left Y  Gaze point right X  Gaze point right Y  ...  \
2339             1094.0               959.0              1034.0  ...   
2340             1091.0               972.0              1040.0  ...   
2341             1097.0               949.0              1051.0  ...   
2342             1101.0               963.0              1072.0  ...   
2343             1088.0               950.0              1055.0  ...   
...                 ...                 ...                 ...  ...   
4226              851.0               727.0               851.0  ...   
4227              869.0               702.0               843.0  ...   
4228              882.0               699.0               853.0  ...   
4230              866.0               724.0               842.0  ...   
4231              866.0               710.0               858.0  ...   

      Gaze point left Y (MCSnorm)  Gaze point right X (MCSnorm)  \
2339                          NaN                        0.4993   
2340                          NaN                        0.5064   
2341                          NaN                        0.4941   
2342                          NaN                        0.5017   
2343                          NaN                        0.4949   
...                           ...                           ...   
4226                       0.7883                        0.3788   
4227                       0.8046                        0.3656   
4228                       0.8170                        0.3641   
4230                       0.8021                        0.3773   
4231                       0.8021                        0.3697   

      Gaze point right Y (MCSnorm)  Eye movement type  \
2339                        0.9573           Fixation   
2340                        0.9628           Fixation   
2341                        0.9728           Fixation   
2342                        0.9923           Fixation   
2343                        0.9769           Fixation   
...                            ...                ...   
4226                        0.7877           Fixation   
4227                        0.7801           Fixation   
4228                        0.7899           Fixation   
4230                        0.7797           Fixation   
4231                        0.7942           Fixati

In [82]:
cleaned_data["視線誘導問題.002"].to_csv("/content/drive/MyDrive/class/情報デザイン論/加工データ csv/user-1/user-1-pic2.csv")

### カラムの要素が1つしかない変数の説明 写真1
|変数名(英語)|変数名(日本語)|中身|
|------------|--------------|----|
|Project name|プロジェクト名|`JD4`|
|Export date|エクスポート日| `2025/07/18`|
|Participant name| 参加者名|`user-1`|
|Recording name| 録画名|`Recording2`|
|Recording date|録画日(JSC) |`2025/07/18`|
|Recording date UTC| 録画日(UTC)|`2025/07/18`|
|Recording start time|録画開始時刻(JSC) |`16:09:36.966`|
|Recording start time UTC| 録画開始時刻(UTC) |`07:09:36.966`|
|Recording duration| 録画時間|`214788`ミリ秒|
|Timeline name| タイムライン名|`Timeline1`|
|Recording Fixation filter name| 録画の注視フィルタ名|`Tobii I-VT (Fixation)`|
|Recording software version| 録画ソフトのバージョン|`24.21.435`|
|Recording resolution height|録画解像度(高さ) |`1080`px|
|Recording resolution width|録画解像度(幅) | `1920`px|
|Recording monitor latency| モニター遅延|`10`ミリ秒|
|Average calibration accuracy | 平均キャリブレーション精度|`17.9`mm|
|Average calibration accuracy | 平均キャリブレーション精度|`1.25`degrees|
|Average calibration accuracy | 平均キャリブレーション精度|`62`pixels|
|Average validation accuracy | 平均バリデーション精度|`8`mm|
|Average validation precision SD | 平均バリデーション予測SD|`3.7`mm|
|Average validation precision RMS |平均バリデーション予測 RMS |`4.9`mm|
|Average validation accuracy |平均バリデーション精度| `0.58`degrees|
|Average validation precision SD |平均バリデーション予測 SD| `0.27`degrees|
|Average validation precision RMS |平均バリデーション予測 RMS| `0.35`degrees|
|Average validation accuracy |平均バリデーション精度| `28`pixels|
|Average validation precision SD |平均バリデーション予測SD| `13`pixels|
|Average validation precision RMS |平均バリデーション予測 RMS| `17`pixels|
|---------------------------------|--------------------------|-----------|
|  Sensor|センサー| `Eye Tracker`|
|  Event|イベント |`nan`|
|Event value|イベントの中身| `nan`|
|Presented Stimulus name|提示された刺激名| `視線誘導問題.001`|
|Presented Media name|提示されたメディア名| `視線誘導問題.001.jpeg`|
|Presented Media width|提示されたメディア幅|`1920`px|
|Presented Media height|提示されたメディア高さ|`1080`px|
|Presented Media position X (DACSpx)|提示されたメディア位置 X|`0`|
|Presented Media position Y (DACSpx)|提示されたメディア位置 Y|`0`|
|Original Media width|オリジナルメディアの幅|`1920`px|
|Original Media height |オリジナルメディアの高さ|`1080`px|
|Mouse position X|マウス位置 X|`nan`|
|Mouse position Y|マウス位置 Y|  `nan`|


In [78]:
cleaned_data["視線誘導問題.001"]

Unnamed: 0  Recording timestamp  Computer timestamp  \
4536        4536             91353604          6496397462   
4537        4537             91370512          6496414370   
4538        4538             91386638          6496430496   
4539        4539             91403573          6496447431   
4540        4540             91420430          6496464288   
...          ...                  ...                 ...   
6963        6963            131741149          6536785007   
6964        6964            131758012          6536801870   
6965        6965            131774442          6536818300   
6967        6967            131791119          6536834977   
6968        6968            131807834          6536851692   

      Eyetracker timestamp  Gaze point X  Gaze point Y  Gaze point left X  \
4536          6.514732e+09        1616.0         575.0             1646.0   
4537          6.514749e+09        1616.0         623.0             1665.0   
4538          6.514765e+09        1573.0         647.0             1613.0   
4539          6.514782e+09        1611.0         570.0             1655.0   
4540          6.514798e+09        1607.0         592.0             1650.0   
...                    ...           ...           ...                ...   
6963          6.555117e+09        1604.0         723.0             1554.0   
6964          6.555134e+09        1533.0         721.0             1535.0   
6965          6.555150e+09        1529.0         707.0             1538.0   
6967          6.555167e+09        1507.0         655.0             1484.0   
6968          6.555183e+09        1486.0         631.0             1503.0   

      Gaze point left Y  Gaze point right X  Gaze point right Y  ...  \
4536              566.0              1586.0               583.0  ...   
4537              603.0              1568.0               644.0  ...   
4538              642.0              1533.0               652.0  ...   
4539              543.0              1568.0               598.0  ...   
4540              576.0              1565.0               608.0  ...   
...                 ...                 ...                 ...  ...   
6963              655.0              1655.0               792.0  ...   
6964              611.0              1531.0               831.0  ...   
6965              589.0              1520.0               824.0  ...   
6967              542.0              1530.0               767.0  ...   
6968              521.0              1469.0               741.0  ...   

      Gaze point left Y (MCSnorm)  Gaze point right X (MCSnorm)  \
4536                       0.5237                        0.8258   
4537                       0.5580                        0.8165   
4538                       0.5947                        0.7983   
4539                       0.5025                        0.8164   
4540                       0.5337                        0.8151   
...                           ...                           ...   
6963                       0.6061                        0.8618   
6964                       0.5653                        0.7976   
6965                       0.5456                        0.7916   
6967                       0.5017                        0.7968   
6968                       0.4828                        0.7651   

      Gaze point right Y (MCSnorm)  Eye movement type  \
4536                        0.5402           Fixation   
4537                        0.5963            Saccade   
4538                        0.6035       Unclassified   
4539                        0.5539            Saccade   
4540                        0.5632           Fixation   
...                            ...                ...   
6963                        0.7331            Saccade   
6964                        0.7693            Saccade   
6965                        0.7629       Unclassified   
6967                        0.7105            Saccade   
6968                        0.6864            Sacca

In [87]:
cleaned_data["視線誘導問題.001"].to_csv("/content/drive/MyDrive/class/情報デザイン論/加工データ csv/user-1/user-1-pic1.csv")

### カラムの要素が1つしかない変数の説明 写真3
|変数名(英語)|変数名(日本語)|中身|
|------------|--------------|----|
|Project name|プロジェクト名|`JD4`|
|Export date|エクスポート日| `2025/07/18`|
|Participant name| 参加者名|`user-1`|
|Recording name| 録画名|`Recording2`|
|Recording date|録画日(JSC) |`2025/07/18`|
|Recording date UTC| 録画日(UTC)|`2025/07/18`|
|Recording start time|録画開始時刻(JSC) |`16:09:36.966`|
|Recording start time UTC| 録画開始時刻(UTC) |`07:09:36.966`|
|Recording duration| 録画時間|`214788`ミリ秒|
|Timeline name| タイムライン名|`Timeline1`|
|Recording Fixation filter name| 録画の注視フィルタ名|`Tobii I-VT (Fixation)`|
|Recording software version| 録画ソフトのバージョン|`24.21.435`|
|Recording resolution height|録画解像度(高さ) |`1080`px|
|Recording resolution width|録画解像度(幅) | `1920`px|
|Recording monitor latency| モニター遅延|`10`ミリ秒|
|Average calibration accuracy | 平均キャリブレーション精度|`17.9`mm|
|Average calibration accuracy | 平均キャリブレーション精度|`1.25`degrees|
|Average calibration accuracy | 平均キャリブレーション精度|`62`pixels|
|Average validation accuracy | 平均バリデーション精度|`8`mm|
|Average validation precision SD | 平均バリデーション予測SD|`3.7`mm|
|Average validation precision RMS |平均バリデーション予測 RMS |`4.9`mm|
|Average validation accuracy |平均バリデーション精度| `0.58`degrees|
|Average validation precision SD |平均バリデーション予測 SD| `0.27`degrees|
|Average validation precision RMS |平均バリデーション予測 RMS| `0.35`degrees|
|Average validation accuracy |平均バリデーション精度| `28`pixels|
|Average validation precision SD |平均バリデーション予測SD| `13`pixels|
|Average validation precision RMS |平均バリデーション予測 RMS| `17`pixels|
|---------------------------------|--------------------------|-----------|
|  Sensor|センサー| `Eye Tracker`|
|  Event|イベント |`nan`|
|Event value|イベントの中身| `nan`|
|Presented Stimulus name|提示された刺激名| `視線誘導問題.003`|
|Presented Media name|提示されたメディア名| `視線誘導問題.003.jpeg`|
|Presented Media width|提示されたメディア幅|`1920`px|
|Presented Media height|提示されたメディア高さ|`1080`px|
|Presented Media position X (DACSpx)|提示されたメディア位置 X|`0`|
|Presented Media position Y (DACSpx)|提示されたメディア位置 Y|`0`|
|Original Media width|オリジナルメディアの幅|`1920`px|
|Original Media height |オリジナルメディアの高さ|`1080`px|
|Mouse position X|マウス位置 X|`nan`|
|Mouse position Y|マウス位置 Y|  `nan`|

In [79]:
cleaned_data["視線誘導問題.003"]

Unnamed: 0  Recording timestamp  Computer timestamp  \
7262        7262            136805014          6541848872   
7263        7263            136821691          6541865549   
7264        7264            136838382          6541882240   
7265        7265            136854931          6541898789   
7266        7266            136871614          6541915472   
...          ...                  ...                 ...   
9362        9362            171789981          6576833839   
9363        9363            171806396          6576850254   
9364        9364            171823249          6576867107   
9366        9366            171839803          6576883661   
9367        9367            171856319          6576900177   

      Eyetracker timestamp  Gaze point X  Gaze point Y  Gaze point left X  \
7262          6.560180e+09         262.0        1106.0              247.0   
7263          6.560197e+09         276.0        1098.0              260.0   
7264          6.560214e+09         278.0        1101.0              265.0   
7265          6.560230e+09         267.0        1113.0              264.0   
7266          6.560247e+09         268.0        1107.0              279.0   
...                    ...           ...           ...                ...   
9362          6.595163e+09        1127.0         876.0             1128.0   
9363          6.595179e+09        1126.0         869.0             1139.0   
9364          6.595196e+09        1130.0         886.0             1136.0   
9366          6.595213e+09        1139.0         893.0             1134.0   
9367          6.595229e+09        1153.0         897.0             1161.0   

      Gaze point left Y  Gaze point right X  Gaze point right Y  ...  \
7262             1160.0               277.0              1053.0  ...   
7263             1159.0               291.0              1037.0  ...   
7264             1154.0               292.0              1047.0  ...   
7265             1173.0               269.0              1053.0  ...   
7266             1162.0               257.0              1051.0  ...   
...                 ...                 ...                 ...  ...   
9362              890.0              1125.0               863.0  ...   
9363              878.0              1114.0               860.0  ...   
9364              896.0              1125.0               876.0  ...   
9366              908.0              1144.0               877.0  ...   
9367              895.0              1146.0               898.0  ...   

      Gaze point left Y (MCSnorm)  Gaze point right X (MCSnorm)  \
7262                          NaN                        0.1440   
7263                          NaN                        0.1517   
7264                          NaN                        0.1520   
7265                          NaN                        0.1401   
7266                          NaN                        0.1339   
...                           ...                           ...   
9362                       0.8241                        0.5862   
9363                       0.8132                        0.5800   
9364                       0.8295                        0.5860   
9366                       0.8411                        0.5961   
9367                       0.8285                        0.5968   

      Gaze point right Y (MCSnorm)  Eye movement type  \
7262                        0.9748           Fixation   
7263                        0.9600           Fixation   
7264                        0.9695           Fixation   
7265                        0.9750           Fixation   
7266                        0.9733           Fixation   
...                            ...                ...   
9362                        0.7988           Fixation   
9363                        0.7959           Fixation   
9364                        0.8111           Fixation   
9366                        0.8117           Fixation   
9367                        0.8319           Fixati

In [86]:
cleaned_data["視線誘導問題.003"].to_csv("/content/drive/MyDrive/class/情報デザイン論/加工データ csv/user-1/user-1-pic3.csv")

### カラムの要素が1つしかない変数の説明 写真4
|変数名(英語)|変数名(日本語)|中身|
|------------|--------------|----|
|Project name|プロジェクト名|`JD4`|
|Export date|エクスポート日| `2025/07/18`|
|Participant name| 参加者名|`user-1`|
|Recording name| 録画名|`Recording2`|
|Recording date|録画日(JSC) |`2025/07/18`|
|Recording date UTC| 録画日(UTC)|`2025/07/18`|
|Recording start time|録画開始時刻(JSC) |`16:09:36.966`|
|Recording start time UTC| 録画開始時刻(UTC) |`07:09:36.966`|
|Recording duration| 録画時間|`214788`ミリ秒|
|Timeline name| タイムライン名|`Timeline1`|
|Recording Fixation filter name| 録画の注視フィルタ名|`Tobii I-VT (Fixation)`|
|Recording software version| 録画ソフトのバージョン|`24.21.435`|
|Recording resolution height|録画解像度(高さ) |`1080`px|
|Recording resolution width|録画解像度(幅) | `1920`px|
|Recording monitor latency| モニター遅延|`10`ミリ秒|
|Average calibration accuracy | 平均キャリブレーション精度|`17.9`mm|
|Average calibration accuracy | 平均キャリブレーション精度|`1.25`degrees|
|Average calibration accuracy | 平均キャリブレーション精度|`62`pixels|
|Average validation accuracy | 平均バリデーション精度|`8`mm|
|Average validation precision SD | 平均バリデーション予測SD|`3.7`mm|
|Average validation precision RMS |平均バリデーション予測 RMS |`4.9`mm|
|Average validation accuracy |平均バリデーション精度| `0.58`degrees|
|Average validation precision SD |平均バリデーション予測 SD| `0.27`degrees|
|Average validation precision RMS |平均バリデーション予測 RMS| `0.35`degrees|
|Average validation accuracy |平均バリデーション精度| `28`pixels|
|Average validation precision SD |平均バリデーション予測SD| `13`pixels|
|Average validation precision RMS |平均バリデーション予測 RMS| `17`pixels|
|---------------------------------|--------------------------|-----------|
|  Sensor|センサー| `Eye Tracker`|
|  Event|イベント |`nan`|
|Event value|イベントの中身| `nan`|
|Presented Stimulus name|提示された刺激名| `視線誘導問題.004`|
|Presented Media name|提示されたメディア名| `視線誘導問題.004.jpeg`|
|Presented Media width|提示されたメディア幅|`1920`px|
|Presented Media height|提示されたメディア高さ|`1080`px|
|Presented Media position X (DACSpx)|提示されたメディア位置 X|`0`|
|Presented Media position Y (DACSpx)|提示されたメディア位置 Y|`0`|
|Original Media width|オリジナルメディアの幅|`1920`px|
|Original Media height |オリジナルメディアの高さ|`1080`px|
|Mouse position X|マウス位置 X|`nan`|
|Mouse position Y|マウス位置 Y|  `nan`|

In [80]:
cleaned_data["視線誘導問題.004"]

Unnamed: 0  Recording timestamp  Computer timestamp  \
9624         9624            176875557          6581919415   
9625         9625            176891913          6581935771   
9626         9626            176908655          6581952513   
9627         9627            176925379          6581969237   
9628         9628            176941836          6581985694   
...           ...                  ...                 ...   
11584       11584            209491704          6614535562   
11585       11585            209508326          6614552184   
11586       11586            209525064          6614568922   
11588       11588            209541627          6614585485   
11589       11589            209558295          6614602153   

       Eyetracker timestamp  Gaze point X  Gaze point Y  Gaze point left X  \
9624           6.600248e+09         423.0        1018.0              473.0   
9625           6.600265e+09         329.0        1081.0                NaN   
9626           6.600281e+09         346.0        1053.0              401.0   
9627           6.600298e+09         340.0        1045.0              397.0   
9628           6.600315e+09         242.0        1095.0              246.0   
...                     ...           ...           ...                ...   
11584          6.632862e+09         745.0         898.0              730.0   
11585          6.632879e+09         755.0         897.0              736.0   
11586          6.632896e+09         763.0         890.0              744.0   
11588          6.632912e+09         765.0         892.0              749.0   
11589          6.632929e+09         752.0         902.0              740.0   

       Gaze point left Y  Gaze point right X  Gaze point right Y  ...  \
9624               855.0               372.0              1181.0  ...   
9625                 NaN               329.0              1081.0  ...   
9626              1021.0               291.0              1085.0  ...   
9627              1002.0               283.0              1088.0  ...   
9628              1113.0               238.0              1078.0  ...   
...                  ...                 ...                 ...  ...   
11584              891.0               759.0               906.0  ...   
11585              888.0               773.0               905.0  ...   
11586              884.0               783.0               895.0  ...   
11588              888.0               781.0               895.0  ...   
11589              902.0               763.0               903.0  ...   

       Gaze point left Y (MCSnorm)  Gaze point right X (MCSnorm)  \
9624                        0.7913                           NaN   
9625                           NaN                           NaN   
9626                        0.9451                           NaN   
9627                        0.9279                           NaN   
9628                           NaN                        0.1239   
...                            ...                           ...   
11584                       0.8252                        0.3953   
11585                       0.8226                        0.4027   
11586                       0.8189                        0.4077   
11588                       0.8223                        0.4066   
11589                       0.8352                        0.3973   

       Gaze point right Y (MCSnorm)  Eye movement type  \
9624                            NaN       Unclassified   
9625                            NaN            Saccade   
9626                            NaN            Saccade   
9627                            NaN       Unclassified   
9628                         0.9977            Saccade   
...                             ...                ...   
11584                        0.8384           Fixation   
11585                        0.8377           Fixation   
11586                        0.8286           Fixation   
11588                        0.8287           Fixat

In [85]:
cleaned_data["視線誘導問題.004"].to_csv("/content/drive/MyDrive/class/情報デザイン論/加工データ csv/user-1/user-1-pic4.csv")